source/grid/grid_generator.cc

```diff
@@ -1910,6 +1910,355 @@ namespace GridGenerator
+             (std::abs(p[1] - c[1]) < radius);
+    }
+
+    // same as above but will ingore the third component
+    // of both points
+    bool inline point_in_2d_box(const Point<3> &p,
+                                const Point<3> &center,
+                                const double    radius)
+    {
+      return point_in_2d_box(Point<2>(p[0], p[1]),
+                             Point<2>(center[0], center[1]),
+                             radius);
+    }
+
+    /**
+     * Keep the main 2D implementation internal as we will use
+     * material id in 3D case, whereas the actual 2D function
+     * will reset material id in addition to this one.
+     */
+    void plate_with_a_hole(Triangulation<2> &       tria,
+                           const double             inner_radius,
+                           const double             outer_radius,
+                           const double             pad_bottom,
+                           const double             pad_top,
+                           const double             pad_left,
+                           const double             pad_right,
+                           const Point<2>           new_center,
+                           const types::manifold_id polar_manifold_id,
+                           const types::manifold_id tfi_manifold_id,
+                           const double             L,
+                           const unsigned int /*n_slices*/,
+                           const bool colorize)
+    {
+      Assert((pad_bottom > 0 || pad_top > 0 || pad_left > 0 || pad_right > 0),
+             ExcMessage("At least one padding parameter has to be non-zero."));
+      Assert(pad_bottom >= 0., ExcMessage("Negative bottom padding."));
+      Assert(pad_top >= 0., ExcMessage("Negative top padding."));
+      Assert(pad_left >= 0., ExcMessage("Negative left padding."));
+      Assert(pad_right >= 0., ExcMessage("Negative right padding."));
+
+      const Point<2> center;
+
+      auto min_line_length = [](const Triangulation<2> &tria) -> double {
+        double length = std::numeric_limits<double>::max();
+        for (const auto cell : tria.active_cell_iterators())
+          for (unsigned int n = 0; n < GeometryInfo<2>::lines_per_cell; ++n)
+            length = std::min(length, cell->line(n)->diameter());
+        return length;
+      };
+
+      // start by setting up the cylinder triangulation
+      Triangulation<2> cylinder_tria;
+      GridGenerator::hyper_cube_with_cylindrical_hole(cylinder_tria,
+                                                      inner_radius,
+                                                      outer_radius,
+                                                      L,
+                                                      /*repetitions*/ 1,
+                                                      colorize);
+
+      // Mark the cylinder material ids as 2 to distinguish them from the bulk
+      // cells (these are read again about 40 lines down)
```

Comment:
A bit pedantic, but these are read again 75 lines down. I would just write "these are read again in the loop that sets manifold ids".